lib/glossarist/concept.rb

```diff
@@ -1,142 +1,225 @@
-# frozen_string_literal: true
+module Glossarist
+  class Concept < Lutaml::Model::Serializable
+    attribute :id, :string
+    attribute :uuid, :string
+    attribute :designations, Designation::Base, collection: true
+    attribute :domain, :string
+    attribute :subject, :string
+    attribute :definition, DetailedDefinition, collection: true
+    attribute :non_verb_rep, :string
+    attribute :notes, DetailedDefinition, default: -> { Glossarist::Collections::Collection.new(klass: DetailedDefinition) }, collection: true
```

Comment:
@HassanAkbar does this mean that we should support "attribute collections" using custom "collection" Ruby class/types instead of just Ruby `Array`s?

Reply:
I think the answer is yes... we need to either provide a `Lutaml::Model::Collection` base class as an interface (e.g. support some methods from Ruby's `Enum`: insert, delete, iterate, etc.)

Reply:
@ronaldtse Sure I'll look into it, I've created an issue for it here -> https://github.com/lutaml/lutaml-model/issues/215